### PR TITLE
OCPCLOUD-1194: Implement cloud-config-file flag in CCCMO render

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Based on https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/56 and https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/48, both are required to merge before that.

Introduces a new flag:
```
Flags:
      --cloud-config-file string             Input path for the cloud config config map file generated by cluster-config-operator.
```

This implementation will unconditionally create a `{destination_dir}/config` directory which would be used as a source of files to copy into `/etc/kubernetes/bootstrap-configs` directory on bootstrap VM.

Based on the information from the `Infrastructure` resource provided from `--cluster-infrastructure-file` the `--cloud-config-file` configMap would be read, unpacked and stored into `{destination_dir}/config/cloud.config`. Each provider explicitly requiring this, would be able to mount `/etc/kubernetes/bootstrap-configs` directory, and read the content from `cloud.conf`. [example](https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/48/files#diff-959463cd4b7144387dc4daa9bf83369e8f20a333faec67872502bbaf594b3739R18) is Azure CCM.

Implementation is [copied](https://github.com/openshift/machine-config-operator/blob/9c6c2bfd7ed498bfbc296d530d1839bd6a177b0b/pkg/operator/bootstrap.go#L322-L341) from MCO which served as a mean to pass cloud config to master ignition file.

To test this PR on Azure, check out installer [PR](https://github.com/openshift/installer/pull/4947), and do:
```bash
cd <your-installer-dir>
./hack/build.sh
OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=quay.io/dgrigore/release:azure-bootstrap ./bin/openshift-install create cluster
```

### CODE REVIEW

**Review by Fedosin — APPROVED** [review]
/approve
/hold

File: cmd/render/main.go [hunk 1]
```diff
@@ -32,6 +33,7 @@ func init() {
 	renderCmd.PersistentFlags().StringVar(&renderOpts.destinationDir, "dest-dir", "", "The destination dir where CCCMO writes the generated static pods for CCM.")
 	renderCmd.PersistentFlags().StringVar(&renderOpts.clusterInfrastructure, "cluster-infrastructure-file", "", "Input path for the cluster infrastructure file.")
 	renderCmd.PersistentFlags().StringVar(&renderOpts.imagesFile, "images-file", "", "Input path for the images config map file.")
+	renderCmd.PersistentFlags().StringVar(&renderOpts.cloudConfigFile, "cloud-config-file", "", "Input path for the cloud config config map file generated by cluster-config-operator.")
```
Comment (elmiko) [inline cmd/render/main.go:36]: could we make this a little more explicit? for example
```suggestion
	renderCmd.PersistentFlags().StringVar(&renderOpts.cloudConfigFile, "cloud-config-file", "", "Input path for the cloud config configMap manifest generated by cluster-config-operator.")
```

**Review by elmiko — COMMENTED** [review]
mostly looks good, i just have a minor suggestion for the help text.

**Comment by Danil-Grigorev** [comment]
@elmiko Added your suggestion. PTAL
/hold cancel

**Review by elmiko — APPROVED** [review]
thanks @Danil-Grigorev !
/lgtm

**Pull request merged**